api/v1alpha1/istiocontrolplane_types.go

```diff
@@ -114,6 +114,23 @@ func NamespacedNameFromRevision(revision string) types.NamespacedName {
 	return nn
 }
 
+type IstioControlPlaneWithProperties struct {
+	*IstioControlPlane
+	Properties IstioControlPlaneProperties
+}
+
+type IstioControlPlaneProperties struct {
+	Mesh *IstioMesh
+}
+
+func (p IstioControlPlaneProperties) GetMesh() *IstioMesh {
+	if p.Mesh != nil {
+		return p.Mesh
+	}
+
+	return nil
```

Comment:
Why not simply?
```suggestion
	return p.Mesh
```
What am I missing?